src/components/Tree/types.ts

```diff
@@ -40,6 +44,32 @@ export type TreeItemProps = {
      * The kinds of dragItems this dropTarget accepts
      */
     accepts?: { within: string | string[]; after: string | string[]; before: string | string[] } | string | string[];
+    children?: ReactNode;
+
+    parentId?: string;
+    level?: number;
+} & (TreeItemWithLabelProps | TreeItemWithContentComponentProps);
+
+export type TreeItemState = {
+    parentId?: string;
+    childrenIds?: string[];
+    level: number;
+    domElement?: HTMLElement;
+};
+
+export type TreeState = {
+    items: Map<string, TreeItemState>;
+    selectedIds: Set<string>;
+    expandedIds: Set<string>;
+    selectionMode: 'single' | 'multiselect';
```

Comment:
multi-select should be allowed or disallowed through property multi-select in the tree component.